scripts/cli.js

```diff
@@ -7,38 +7,51 @@ const Kredits = require('../lib/kredits');
 module.exports = function(callback) {
   const Registry = artifacts.require('./Registry.sol');
   Registry.deployed().then(async (registry) => {
-    let contractName = await promptly.prompt('Contract Name: ');
-    let method = await promptly.prompt('Function: ');
-    let argumentInput = await promptly.prompt('Arguments (comma separated): ', { default: '' });
-    let args = [];
-    if (argumentInput !== '') {
-      args = argumentInput.split(',').map(a => a.trim());
-    }
-
     const networkId = parseInt(web3.version.network);
     const provider = new ethers.providers.Web3Provider(
       web3.currentProvider, { chainId: networkId }
     );
     const kredits = await Kredits.setup(provider, provider.getSigner());
 
-    const contract = kredits[contractName].contract;
-    console.log(`Using ${contractName} at ${contract.address}`);
-    console.log(`Calling ${method} with ${JSON.stringify(args)}`);
+    let contractName = await promptly.prompt('Contract Name: ');
+    const contractWrapper = kredits[contractName];
 
-    if (!contract[method]) {
+    let method;
+    method = await promptly.prompt('Function (? for available functions): ');
+    while (method === '?') {
+      console.log(`Contract functions: ${JSON.stringify(Object.keys(contractWrapper.functions))}`);
+      console.log(`\nWrapper functions: ${JSON.stringify(Object.getOwnPropertyNames(Object.getPrototypeOf(contractWrapper)))}`);
+      console.log("\n");
+
+      method = await promptly.prompt('Function: ');
+    }
+    if (!contractWrapper[method] && !contractWrapper.functions[method]) {
       callback(new Error(`Method ${method} is not defined on ${contractName}`));
       return;
     }
+    let argumentInput = await promptly.prompt('Arguments (comma separated): ', { default: '' });
+    let args = [];
+    if (argumentInput !== '') {
+      args = argumentInput.split(',').map(a => a.trim());
```

Comment:
Haha master of es6. I think you don’t need the if and the initial `args`
`let args = argumentInput.split(',')` should return an empty array.

Reply:
oh yeah, true.

Reply:
ah noo. it returns an array with an empty string, that was the reason for the if. or what am I missing?

Reply:
Ah ok.

Reply:
But if input is already not an empty string, wouldn't it return an array with one argument string?

Reply:
I was actually asking about the `trim` inside, but no idea why. It's for removing whitespace, so that's still needed.

Reply:
yep, trim() for potential whitespace as for example in `"first, second"`